server/src/storage/data_transfer.rs

```diff
@@ -0,0 +1,399 @@
+        assert!(target_dir.path().join(format!("{}/compressed/0-3.parquet", KEY)).exists());
+        assert_eq!(*data_transfer.compressed_files.get(&KEY).unwrap(), 0 as usize);
+    }
+
+    #[test]
+    fn test_transfer_multiple_files() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let path_1 = create_compressed_file(temp_dir.path(), "test_1");
+        let path_2 = create_compressed_file(temp_dir.path(), "test_2");
+
+        data_transfer.add_compressed_file(&KEY, path_1.as_path()).unwrap();
+        data_transfer.add_compressed_file(&KEY, path_2.as_path()).unwrap();
+        data_transfer.transfer_data(&KEY).unwrap();
+
+        assert!(!path_1.exists());
+        assert!(!path_2.exists());
+
+        let target_path = target_dir.path().join(format!("{}/compressed/0-3.parquet", KEY));
+        assert!(target_path.exists());
+        assert_eq!(*data_transfer.compressed_files.get(&KEY).unwrap(), 0 as usize);
+
+        // The transferred file should have 6 rows since each compressed file has 3.
+        let batch = StorageEngine::read_entire_apache_parquet_file(target_path.as_path()).unwrap();
+        assert_eq!(batch.num_rows(), 6);
+    }
+
+    #[test]
+    fn test_transfer_if_reaching_batch_size_when_adding() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let parquet_path = create_compressed_file(temp_dir.path(), "test");
+
+        // Set the max batch size to ensure that the file is transferred immediately.
+        data_transfer.transfer_batch_size_in_bytes = COMPRESSED_FILE_SIZE - 1;
+        data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).unwrap();
+
+        assert!(target_dir.path().join(format!("{}/compressed/0-3.parquet", KEY)).exists());
+        assert_eq!(*data_transfer.compressed_files.get(&KEY).unwrap(), 0 as usize);
+    }
+
+    #[test]
+    fn test_transfer_if_reaching_batch_size_on_start_up() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        create_compressed_file(temp_dir.path(), "test_1");
+        create_compressed_file(temp_dir.path(), "test_2");
+        create_compressed_file(temp_dir.path(), "test_3");
+
+        // Since the max batch size is 1 byte smaller than 3 compressed files, the data should be transferred immediately.
+        let (target_dir, data_transfer) = create_data_transfer_component(temp_dir.path());
+
+        assert!(target_dir.path().join(format!("{}/compressed/0-3.parquet", KEY)).exists());
+        assert_eq!(*data_transfer.compressed_files.get(&KEY).unwrap(), 0 as usize);
+    }
+
+    /// Set up a data folder with a key folder that has a single compressed file in it.
+    /// Return the path to the created Apache Parquet file.
+    fn create_compressed_file(data_folder_path: &Path, file_name: &str) -> PathBuf {
```

Comment:
Maybe use `local_data_folder_path` for consistency with `DataFolders` in `main.rs`.